src/routes.js

```diff
@@ -494,7 +494,28 @@ const routes = [
       }
     }
   }),
-
+  asyncRoute({
+    path: '/projects/:projectId([1-9]\\d*)/datasets/:datasetName',
+    component: 'DatasetShow',
+    props: true,
+    loading: 'page',
+    key: ({ projectId, datasetName }) =>
+      `/projects/${projectId}/datasets/${encodeURIComponent(datasetName)}`,
+    children: [
+      asyncRoute({
+        path: '',
+        component: 'DatasetOverview',
```

Comment:
It'd be great to add a test that a user who is not logged in is redirected to login if they try to navigate to this route. (Look for `requireLogin` in test/router.spec.js.)